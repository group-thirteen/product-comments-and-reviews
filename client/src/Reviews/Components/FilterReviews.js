
import React from 'react'

class FilterReviews extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      selection: 0
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.handleRatingSelection(Number(event.target.value))
  }

  render () {
    return (
      <form>
        <label>
          <select value={this.state.value} onChange={this.handleChange} >
            <option value={0}>star rating</option>
            <option value={1}>One</option>
            <option value={2}>Two</option>
            <option value={3}>Three</option>
            <option value={4}>Four</option>
            <option value={5}>Five</option>
          </select>
        </label>
      </form>
    )
  }
}

export default FilterReviews
