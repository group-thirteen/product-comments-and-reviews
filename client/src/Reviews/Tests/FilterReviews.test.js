
import React from 'react'
import { shallow } from 'enzyme'
import FilterReviews from '../Components/FilterReviews'
import { createSerializer } from 'enzyme-to-json'
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))


describe('Filter Reviews', () => {
  const handleRatingSelectionSpy = jest.fn()
  const wrapped = shallow(<FilterReviews handleRatingSelection={handleRatingSelectionSpy}/>)
  const select = wrapped.find('select')
  select.simulate('change', {target: {value: 1}})
  it('should call handleRatingSelection', () => {
    expect(handleRatingSelectionSpy).toHaveBeenCalledWith(1)
  })
  it('should render correctly', () => {
    expect(wrapped).toMatchSnapshot()
  })
})
