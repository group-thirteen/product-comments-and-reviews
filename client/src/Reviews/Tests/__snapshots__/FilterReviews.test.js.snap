// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Reviews should render correctly 1`] = `
<form
  className="filter-form"
>
  <label>
    <select
      onChange={[Function]}
    >
      <option
        value={0}
      >
        star rating
      </option>
      <option
        value={1}
      >
        One
      </option>
      <option
        value={2}
      >
        Two
      </option>
      <option
        value={3}
      >
        Three
      </option>
      <option
        value={4}
      >
        Four
      </option>
      <option
        value={5}
      >
        Five
      </option>
    </select>
  </label>
</form>
`;
